terraform/modules/networking/vpc.tf

```diff
@@ -84,3 +84,25 @@ data "aws_iam_policy_document" "vpc_flow_logs_assume_role_permissions" {
     ]
   }
 }
+
+data "aws_vpc" "default_eu_west_2" {
+  default = true
+}
+
+data "aws_vpc" "default_eu_west_1" {

```

Comment:
I tried to define a local variable `non_default_providers = [aws.eu-west-1, aws.eu-west-3, aws.us-east-1]` and then use a single `aws_vpc` data block for these 3 non-default providers (using `count = length(local.non_default_providers)`) but this didn't work - Terraform complained about the use of a `default` attribute when I tried this. So I've left as separate data blocks, which I think is fine.

Reply:
Do you mean count there or do you mean a for_each looping over the non default providers?

Reply:
I had a go with `for_each` but couldn't get it working, and in the end concluded that it's not possible to do this (based on  some google searches) because terraform treats providers in a different way to regular resources/attributes.